HomeWorks/src/Lesson4/Person.java

```
@@ -1,30 +1,39 @@
 package Lesson4;
 
+import Lesson4.exeptions.InvalidNamingException;
+
+import java.util.Locale;
 import java.util.Random;
 
 public class Person {
     private Address address;
     private int age;
     private String name;
+    private String surname;
     private int height;
     private String sex;
 
     public Person() {
     }
 
-    public Person(int age, String name, int height, String sex) {
-        this(age, name, sex);
+    public Person( String name, String surname, int age, int height, String sex) {
+        this.age = age;
+        try{
+            name = editString(name);
+            this.name = name;
+        } catch (InvalidNamingException e) {
+            System.out.println("При создании человека было неверно указано имя");
```

i think you don't want to catch it here since we don't want to allow the creation of Person with wrong name. You should throw this exception further